### ISSUE TITLE
aws S3 ResourceDownloader failure on any model

### ISSUE THREAD

**Issue by marilenaoita** [JohnSnowLabs/spark-nlp#640]
wrt the proxy all in configured , but either in java or python I get this error when Im trying to use the models, either simple pipeline components or pretrained pipeline models. 

e.g., 
PretrainedPipeline pipeline = new PretrainedPipeline("explain_document_ml", "en", "public/models,2.2.1");

or
LemmatizerModel lemmatizer = (LemmatizerModel) ((LemmatizerModel) LemmatizerModel.pretrained("lemma_antbnc", "en", "public/models").setInputCols(new String[] { "normalized" })).setOutputCol("lemma");

## Description

Exception in thread "main" java.lang.NullPointerException
	at com.amazonaws.ClientConfiguration.getProxyUsernameEnvironment(ClientConfiguration.java:874)
	at com.amazonaws.ClientConfiguration.getProxyUsername(ClientConfiguration.java:902)
	at com.amazonaws.http.settings.HttpClientSettings.getProxyUsername(HttpClientSettings.java:90)
	at com.amazonaws.http.settings.HttpClientSettings.isAuthenticatedProxy(HttpClientSettings.java:182)
	at com.amazonaws.http.apache.client.impl.ApacheHttpClientFactory.addProxyConfig(ApacheHttpClientFactory.java:96)
	at com.amazonaws.http.apache.client.impl.ApacheHttpClientFactory.create(ApacheHttpClientFactory.java:75)
	at com.amazonaws.http.apache.client.impl.ApacheHttpClientFactory.create(ApacheHttpClientFactory.java:38)
	at com.amazonaws.http.AmazonHttpClient.<init>(AmazonHttpClient.java:324)
	at com.amazonaws.http.AmazonHttpClient.<init>(AmazonHttpClient.java:308)
	at com.amazonaws.AmazonWebServiceClient.<init>(AmazonWebServiceClient.java:229)
	at com.amazonaws.AmazonWebServiceClient.<init>(AmazonWebServiceClient.java:181)
	at com.amazonaws.services.s3.AmazonS3Client.<init>(AmazonS3Client.java:617)
	at com.amazonaws.services.s3.AmazonS3Client.<init>(AmazonS3Client.java:597)
	at com.amazonaws.services.s3.AmazonS3Client.<init>(AmazonS3Client.java:575)
	at com.amazonaws.services.s3.AmazonS3Client.<init>(AmazonS3Client.java:542)
	at com.johnsnowlabs.nlp.pretrained.S3ResourceDownloader.client$lzycompute(S3ResourceDownloader.scala:45)
	at com.johnsnowlabs.nlp.pretrained.S3ResourceDownloader.client(S3ResourceDownloader.scala:36)
	at com.johnsnowlabs.nlp.pretrained.S3ResourceDownloader.downloadMetadataIfNeed(S3ResourceDownloader.scala:69)
	at com.johnsnowlabs.nlp.pretrained.S3ResourceDownloader.resolveLink(S3ResourceDownloader.scala:81)
	at com.johnsnowlabs.nlp.pretrained.S3ResourceDownloader.getDownloadSize(S3ResourceDownloader.scala:159)
	at com.johnsnowlabs.nlp.pretrained.ResourceDownloader$.getDownloadSize(ResourceDownloader.scala:382)
	at com.johnsnowlabs.nlp.pretrained.ResourceDownloader$.downloadResource(ResourceDownloader.scala:303)
	at com.johnsnowlabs.nlp.pretrained.ResourceDownloader$.downloadModel(ResourceDownloader.scala:343)
	at com.johnsnowlabs.nlp.pretrained.ResourceDownloader$.downloadModel(ResourceDownloader.scala:338)
	at com.johnsnowlabs.nlp.annotators.PretrainedLemmatizer$class.pretrained(LemmatizerModel.scala:41)
	at com.johnsnowlabs.nlp.annotators.LemmatizerModel$.pretrained(LemmatizerModel.scala:44)
	at com.johnsnowlabs.nlp.annotators.LemmatizerModel.pretrained(LemmatizerModel.scala)
	at spark_nlp_pubmed.SparkParallelIndexing.test2(SparkParallelIndexing.java:106)
	at spark_nlp_pubmed.SparkParallelIndexing.main(SparkParallelIndexing.java:130)

## Steps to Reproduce

1.
2.
3.
4.

## Your Environment

* Version used: 2.2.1 spark-nlp and spark 2.4.4 
* Browser Name and version:
* Operating System and version (desktop or mobile):
* Link to your project:

*Any advice on how can I solve it would be really appreciated, thanks ! *

**Comment by maziyarpanahi** [comment]
There seems to be a problem with systems behind a proxy and accessing the public S3. We still don't know why and whether it's up to us or the user's configuration, but you can download the offline models/pipelines and simply load them until this gets fixed:

https://nlp.johnsnowlabs.com/docs/en/pipelines
https://nlp.johnsnowlabs.com/docs/en/models

**Comment by vkocaman** [comment]
Just catching up on the two errors for pretrained() and .load().
	Spark NLP uses ~/cache_pretrained to download models/pipelines and extracts them to load them right after. This is in the /home/username on local and hdfs:///user/username in a cluster. The home directory is always being mounted with write permission and it is executable. (this can be changed in application.conf to point to some place else)
	Spark NLP uses /tmp for lots of TF related ops. The one you guys faced was the very first which TensorFlow automatically puts the .so files there. (nobody can change this). We also do lots of extract and compress via /tmp directory, which this one is also not changeable not even via Apache Spark default dir. The /tmp directory is also mostly writeable and executable (it mounts like this by default). Some people for some strange reason think removing or restricting those permissions bring some sort of security, but after all these years it has been proven otherwise and just messes with lots of dependencies. software, packages, etc. Make sure you check in /etc/fstab that the /tmp is being mounted with write and exec permissions.
	The other error for pretrained() is that it is behind the corporate proxy/firewall. Either the have to whitelist our S3 or the proxy have to be set manually. (the first option is much easier, but proxy/firewall usually means tight security for good reasons, so this means Fat JARs and already downloaded models/pipelines for total offline access)

 & a working solution for Linux machines:

>> mount /tmp -o remount,exec
